src/main/io/gps.c

```diff
@@ -1578,9 +2121,46 @@ static bool UBLOX_parse_gps(void)
     uint32_t i;
 
     *gpsPacketLogChar = LOG_IGNORED;
+#define CLSMSG(cls, msg) (((cls) << 8) | (msg))
+    switch (CLSMSG(_class, _msg_id)) {
+    case CLSMSG(CLASS_MON, MSG_MON_VER): {
+        char *token = strtok(_buffer.ver.swVersion, " ");
+        while (token != NULL) {
+            float f = fastA2F(token);
+            if (f > 0.0f) {
+                int major = strtol(token, NULL, 10);
+                int minor = strtol(&token[major > 9 ? 3 : 2], NULL, 10);
+                gpsData.monVer.swVersion.firmwareVersion.major = major;
+                gpsData.monVer.swVersion.firmwareVersion.minor = minor;
+                break;
+            }
+            token = strtok(NULL, " ");
+        }
+        size_t j = 0;
+        while (j < 10) {
+            if (_buffer.ver.extension[j][0] == '\0') {
+                break;
+            }
+            token = strtok(&_buffer.ver.extension[j++][0], "=");
+            while (token != NULL) {
+                if (strcmp(token, "PROTVER") == 0) {
+                    token = strtok(NULL, "=");
+                    int major = strtol(token, NULL, 10);
+                    int minor = strtol(&token[major > 9 ? 3 : 2], NULL, 10);
+                    gpsData.monVer.swVersion.protocolVersion.major = major;
+                    gpsData.monVer.swVersion.protocolVersion.minor = minor;
+                    break;
+                }
+                token = strtok(NULL, "=");
+            }
+        }
```

Comment:
Token parsing is quite fragile. I'd prefer to rewrite it (I posted some alternatives in comments)

Reply:
@ledvinap freasy is very busy with other commitments right now.  I'd welcome a code suggestion as to how to improve this section; please make your suggestions very explicit, since I am not familiar with 'token' coding at all.

Reply:
this code has been removed